diff -up perl-5.10.1/sv.c.orig perl-5.10.1/sv.c
--- perl-5.10.1/sv.c.orig	2016-07-20 17:15:29.276540420 -0300
+++ perl-5.10.1/sv.c	2016-07-20 17:17:32.706773092 -0300
@@ -8057,9 +8057,14 @@ Perl_sv_pvn_force_flags(pTHX_ SV *sv, ST
 		Perl_croak(aTHX_ "Can't coerce readonly %s to string", ref);
 	}
 	if ((SvTYPE(sv) > SVt_PVLV && SvTYPE(sv) != SVt_PVFM)
-	    || isGV_with_GP(sv))
-	    Perl_croak(aTHX_ "Can't coerce %s to string in %s", sv_reftype(sv,0),
-		OP_NAME(PL_op));
+	    || isGV_with_GP(sv)) {
+	    const char * const ref = sv_reftype(sv,0);
+	    if (PL_op)
+		Perl_croak(aTHX_ "Can't coerce %s to string in %s",
+			   ref, OP_NAME(PL_op));
+	    else
+		Perl_croak(aTHX_ "Can't coerce %s to string", ref);
+	}
 	s = sv_2pv_flags(sv, &len, flags);
 	if (lp)
 	    *lp = len;
