diff -Naur old/dist/Math-BigInt/lib/Math/BigInt.pm new/dist/Math-BigInt/lib/Math/BigInt.pm
--- old/dist/Math-BigInt/lib/Math/BigInt.pm	2011-09-03 16:26:41.000000000 +1000
+++ new/dist/Math-BigInt/lib/Math/BigInt.pm	2013-04-15 14:39:17.890447159 +1000
@@ -60,8 +60,6 @@
 '>>='	=>	sub { $_[0]->brsft($_[1]); },
 
 # not supported by Perl yet
-'..'	=>	\&_pointpoint,
-
 '<=>'	=>	sub { my $rc = $_[2] ?
                       ref($_[0])->bcmp($_[1],$_[0]) : 
                       $_[0]->bcmp($_[1]); 
