diff -up perl-5.10.1/ext/Test-Harness/bin/prove.fork perl-5.10.1/ext/Test-Harness/bin/prove
--- perl-5.10.1/ext/Test-Harness/bin/prove.fork	2011-01-05 13:02:44.050259112 +0100
+++ perl-5.10.1/ext/Test-Harness/bin/prove	2011-01-05 13:03:06.505268159 +0100
@@ -33,7 +33,6 @@ Boolean options:
       --ext         Set the extension for tests (default '.t')
  -f,  --failures    Show failed tests.
  -o,  --comments    Show comments.
-      --fork        Fork to run harness in multiple processes.
       --ignore-exit Ignore exit status from test scripts.
  -m,  --merge       Merge test scripts' STDERR with their STDOUT.
  -r,  --recurse     Recursively descend into directories.
@@ -76,7 +75,7 @@ in F<.proverc> are specified in the same
 
     # .proverc
     --state=hot,fast,save
-    -j9 --fork
+    -j9
 
 Additional option files may be specified with the C<--rc> option.
 Default option file processing is disabled by the C<--norc> option.
